File: recursion-tutorial-fc/reverseString.js
const reverseString = (string) => {

    if(string == null || !string) return "";

    const result = reverseString(string.substring(1)) + string.charAt(0);
    return result;
}


const reverseString2 = (string) => {

    let result = "";

    for(let i = string.length - 1; i >= 0; i--){

        result += string[i];
    }

    return result;
}

console.log(reverseString("Hello"));
console.log(reverseString2("Hello"));

File: recursion-tutorial-fc/ReverseString.java
public class ReverseString {

    public static String reverse(String in) {
        
        if(in.equals("")) return "";

        return reverse(in.substring(1)) + in.charAt(0);
    }


    public static void main(String[] args){

        System.out.println(ReverseString.reverse("Hello"));
        
        //""
        //o + l
        //lo + l
        //llo + e
        //ello + H
        //Hello
    }
}